spec/src/main/doc/javaee/definition_ee.asciidoc

```diff
@@ -0,0 +1,89 @@
+
+When running in Java EE, the container must extend the concepts defined in <<concepts>> with:
+
+* A Java EE component is a _bean_ if the lifecycle of its instances may be managed by the container according to the lifecycle context model defined in <<contexts>>
+* Contextual instances of a bean may be used in EL expressions that are evaluated in the same context
+* For some Java EE components - like environment resources, defined in <<resources>>  - the developer provides only the annotations and the bean implementation is provided by the container.
+
+[[capabilities_ee]]
+
+=== Functionality provided by the container to the bean in Java EE
+
+When running in Java EE, the container must extend the capabilities defined in <<capabilities>>, by providing:
+
+* scoped resolution by bean name when used in a Unified EL expression, as defined by <<name_resolution>>.
+
+[[bean_types_ee]]
+
+=== Bean types for Java EE component
+
+As managed beans, EJB session beans may have multiple bean types depending on their client-visible types. For instance, this session bean has only the local interfaces +BookShop+ and +Auditable+, along with +Object+, as bean types, since the bean class is not a client-visible type.
+                                                                                                                        
+[source, java]
+----
+@Stateful 
+public class BookShopBean 
+        extends Business 
+        implements BookShop, Auditable { 
+    ... 
+}
+----
+
+The rules for determining the (unrestricted) set of bean types for Java EE components are defined in  <<session_bean_types>> and <<resource_types>>.
+
+[[scopes_ee]]
+
+=== Scopes
+
+Java EE components such as servlets, EJBs and JavaBeans do not have a well-defined _scope_. These components are either:
+
+* _singletons_, such as EJB singleton session beans, whose state is shared between all clients,
+* _stateless objects_, such as servlets and stateless session beans, which do not contain client-visible state, or
+* objects that must be explicitly created and destroyed by their client, such as JavaBeans and stateful session beans, whose state is shared by explicit reference passing between clients.
+
+CDI scopes add to Java EE these missing well-defined lifecycle context as defined in <<scopes>>.
+
+[[builtin_scopes_ee]]
+
+==== Built-in scope types in Java EE
+
+When running in Java EE, the implementation of the @RequestScoped, @ApplicationScoped and @SessionScoped annotations provided by the container, represents the standard scopes defined by the Java Servlets specification.
+
+
+[[default_bean_discovery_ee]]
+=== Default bean discovery mode for Java EE
+
+When running in Java EE, the container must extend the rules defined in <<default_bean_discovery>> with:
+
+If the _bean discovery mode_ is `annotated`, then:
+
+* bean classes of EJB sessions beans, are not discovered, and
+* producer methods that are not on an EJB session bean are not discovered, and
```

Comment:
This is confusing as it kind of implies that in `annotated` archive in Java EE a producer/disposer/observer has to be defined on an EJB session bean